main/ertp/ERTP-Guide/Overview.md

```diff
@@ -0,0 +1,276 @@
+
+ERTP `purse` objects have a `deposit` method which takes a `payment`
+object as its argument. It first checks that the `payment` object is
+genuine and the same asset type as the `purse` (after all, our quatloos currency is
+a very unstable asset, so we wouldn't want to let people deposit
+quatloos and get credit for simoleans). Note: quatloos and simoleans
+are both imaginary currencies. 
+
+If everything passes the checks, the asset moves from 
+the `payment` to the `purse`. If there's a problem, it throws an error.
+
+After a successful deposit, ERTP guarantees:
+- The `payment` object is empty.
+- The `purse` contains the `payment`'s full content.
+
+When the `deposit` call throws an error (i.e. something went wrong), ERTP guarantees:
+
+- The alleged `payment` is in the same state as before the call.
+- The `purse` is in the same state as before the call.
+
+## Promises
+
+Several ERTP methods are *asynchronous* and instead of immediately returning their expected value, return a *promise* for that value.
+
+JavaScript implements `Promise` objects, and recently added the two commands `async` and `await` to simplify working with them. For general, and extensive, information about JavaScript's implementation, see either:
+- [javascript.info](https://javascript.info/async)
+- [Mozilla's Developer Docs](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous)
+
+**tyg todo: Needs beefing up about how Promises, async, and await are
+  actually used in contracts**
+
+
+## Alleged Names
+
+ERTP implements [*object capabilities*](https://agoric.com/documentation/glossary/#object-capabilities). You can only use an object and issue commands to it if you have access to that object, not just its human-readable name or similar. For example, I might know (or make a good guess), that the mint that makes quatloos has the human-understandable name of 'quatloos-mint'. But unless I have the actual object that is a `mint` object associated with the `quatloos` `brand` object, I can't use it to create a million `quatloos` and bet
+them all on Captain Kirk to win his gladitorial match on Triskelion (see the[ Wikipedia entry](https://en.wikipedia.org/wiki/The_Gamesters_of_Triskelion)).
+
+**tyg todo: Needs beefing up on how allegedNames are used in contracts
+  vs. getting and using the real objects**
+
+## Creating and sending a fungible asset
+
+In ERTP, digital assets are created by a [`mint`](). You must have
+access to a `mint` object to create new assets of the `mint` object's
+associated `brand`. 
+
+For instance, let's say we want to create a new community
+currency called 'BaytownBucks'. You start by getting access to a `mint`
+object whose associated brand is 'BaytownBucks'.
+```js
+import { makeMint } from '@agoric/ertp';
+const baytownBucksMint = makeMint('BaytownBucks');
+```
+
+Now, use the `mint` to create 1000 new BaytownBucks.
+
+```js
+const { issuer, mint } = produceIssuer('BaytownBucks');
+const BB1000 = amountMath.make(1000);
```

Comment:
We shouldn't use variable names with starting capitals in JavaScript. That is usually reserved for classes